src/platform/backends/shared/base_virtual_machine_factory.h

```diff
@@ -34,17 +35,23 @@ class BaseVirtualMachineFactory : public VirtualMachineFactory
 {
 public:
     BaseVirtualMachineFactory() = default;
```

Comment:
Are these here because of unit tests? I didn't touch them as I'm unfamiliar with how they work or why exactly they're required...

Reply:
OK, this is another case like the previous one. Constructors are meant to ensure objects are properly initialized. On top of that, tests should not be using dedicated code paths that are separate from production. Uninitialized/stub bits are the responsibility of stub implementations, which should use the same constructors as anyone else.

This constructor was correct before, because there was nothing to initialize. But the parameter you now need should go in this constructor, not in a new one. 

I added a commit on top to get rid of most of these stub constructors. I left one (pre-existing) constructor in QEMU as an exercise for later.